main.go

```diff
@@ -370,7 +373,37 @@ func main() {
 		}
 	}
 	// handle profiler sink close on exit
-	defer prfl.Stop()
+	defer func() {
+		logger.Infow("closing profiler sink client before exiting", "sink", prfl.GetSinkName())
+		prfl.Stop()
+	}()
+
+	// tracer sink
+	ts, err := tracer.GetSink(logger, tracertypes.SinkDriver(cfg.Controller.TracerSink))
+	if err != nil {
+		logger.Errorw("error while creating profiler sink, switching to noop", "error", err)
+
+		ts, err = tracer.GetSink(logger, tracertypes.SinkDriverNoop)
+
+		if err != nil {
```

Comment:
I mean... Can this happen?

If not, maybe let's have a NewNoop that returns no error (all)

Reply:
> I mean... Can this happen?

As things are, no. This is more of a interface safety so we never end with a potentially untracked error if the `noop` sink ever returns an error -- were we to ever change that code. Happy to ignore that error if you think it's useless / too much / should be modified with the noop sink if needed.

> maybe let's have a NewNoop that returns no error (all)

I like the clarity of consistently using the same function (`tracer.GetSink(...)`) for every tracer init, wether it's the `noop` sink or another. I believe this is a small price for cleaner, more readable `tracer` package entry point.

Reply:
For this PR I'll remove the noop sink error handling -- I get it is indeed a little bit much.
I'd love to discuss this with you and change the Factory pattern across metrics, profiler, tracer in a separate PR.